src/virtual_secondary/managedsecondary.cc

```diff
@@ -72,6 +73,10 @@ void ManagedSecondary::rawToMeta() {
 }
 
 bool ManagedSecondary::putMetadata(const Uptane::RawMetaPack &meta_pack) {
+  if (fiu_fail("secondary_putmetadata") != 0) {
```

Comment:
I suppose it's fine in the given context to inject into the "production" code some test code but IMHO, implementation of SecondaryInterface that realizes specific test case and is instantiated within a corresponding test and then just aktualizr->AddSecondary would do more robust and flexible. Otherwise, adding a new test case means adding another `if (fiu_fail("") != 0)` into the "production" code.

Reply:
Yes that's a valid point. I've chosen the fiu solution one part because it's simpler and it's also easier to trigger the error against the real backend by scheduling an update and launching aktualizr with fiu.

For now, the code clutter seem minimal enough to me but I understand that's subjective.

Reply:
+1:

Reply:
Can you get rid of this `fiu_fail` bit now that it's in `VirtualSecondary`?

Reply:
Oops I thought I did. Thanks!